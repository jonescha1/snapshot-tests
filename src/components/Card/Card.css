@import url("https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&display=swap");
* {
  margin: 0;
  padding: 0;
  font-family: "Montserrat", sans-serif;
  color: rgb(90, 87, 87);
}
.card {
  position: relative;
  width: 375px;
  height: 500px;
  margin: auto;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 4px 12px;
  border-radius: 50px;
  background-color: #f1f1f1;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.background-img {
  height: 40%;
  width: 100%;
  filter: blur(1px);
}

.content {
  position: relative;
  top: -87.5px;
  max-width: 86%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.name {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.name h2 {
  margin-top: 20px;
}

.aboutme {
  font-size: 0.8em;
  margin-top: 20px;
  font-weight: 400;
  color: gray;
}

.social-links {
  display: flex;
  width: 100%;
  justify-content: space-evenly;
  margin-top: 15px;
}

.profile-pic {
  width: 175px;
  border-radius: 50%;
  z-index: 1;
}
.profile-pic-backdrop {
  background-color: #f1f1f1;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 4px 12px;
  width: 200px;
  height: 200px;
  border-radius: 50%;
  position: absolute;
  top: -12px;
}
