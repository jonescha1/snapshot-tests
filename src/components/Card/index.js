import React from "react";
import "./Card.css";

import BackgroundImg from "../../images/mike-scheid-xoYPV4oVQJI-unsplash.jpg";
import ProfileImg from "../../images/profile_pic_chase.jpeg";

import "boxicons";

function Card() {
  return (
    <div className="card">
      <img
        className="background-img"
        src={BackgroundImg}
        alt="card background"
      />
      <div className="content">
        <img className="profile-pic" src={ProfileImg} alt="Users profile" />
        <div className="profile-pic-backdrop" />
        <div className="name">
          <h2>Chase Jones</h2>
          <h5>@chaseJones</h5>
        </div>
        <p className="aboutme">
          I am a current Full Stack Developer Student at NC State University. I
          enjoy spending time with my family and friends, CODING, playing guitar
          and Hunting.
        </p>
        <div className="social-links">
          <a href="https://github.com/jonescha1">
            <box-icon type="logo" name="github"></box-icon>
          </a>
          <a href="https://www.facebook.com/chase.jones.9465">
            <box-icon type="logo" name="facebook"></box-icon>
          </a>
          <a href="https://www.linkedin.com/in/chase-jones-b0a163223/">
            <box-icon name="linkedin-square" type="logo"></box-icon>
          </a>
        </div>
      </div>
    </div>
  );
}

export default Card;
