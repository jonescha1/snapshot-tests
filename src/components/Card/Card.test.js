import renderer from "react-test-renderer";
import Card from "./index";

test("renders a snapshot of the card", () => {
  const tree = renderer.create(<Card />).toJSON();

  expect(tree).toMatchSnapshot();
});
