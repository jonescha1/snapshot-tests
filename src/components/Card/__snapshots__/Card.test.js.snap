// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of the card 1`] = `
<div
  className="card"
>
  <img
    alt="card background"
    className="background-img"
    src="mike-scheid-xoYPV4oVQJI-unsplash.jpg"
  />
  <div
    className="content"
  >
    <img
      alt="Users profile"
      className="profile-pic"
      src="profile_pic_chase.jpeg"
    />
    <div
      className="profile-pic-backdrop"
    />
    <div
      className="name"
    >
      <h2>
        Chase Jones
      </h2>
      <h5>
        @chaseJones
      </h5>
    </div>
    <p
      className="aboutme"
    >
      I am a current Full Stack Developer Student at NC State University. I enjoy spending time with my family and friends, CODING, playing guitar and Hunting.
    </p>
    <div
      className="social-links"
    >
      <a
        href="https://github.com/jonescha1"
      >
        <box-icon
          name="github"
          type="logo"
        />
      </a>
      <a
        href="https://www.facebook.com/chase.jones.9465"
      >
        <box-icon
          name="facebook"
          type="logo"
        />
      </a>
      <a
        href="https://www.linkedin.com/in/chase-jones-b0a163223/"
      >
        <box-icon
          name="linkedin-square"
          type="logo"
        />
      </a>
    </div>
  </div>
</div>
`;
